components/script/dom/bindings/str.rs

```diff
@@ -182,16 +182,20 @@ impl DOMString {
             return;
         }
 
-        let last_non_whitespace = match self.0.rfind(|ref c| !char::is_ascii_whitespace(c)) {
-            Some(idx) => idx + 1,
+        match self.0.rfind(|ref c| !char::is_ascii_whitespace(c)) {
+            Some(idx) => {
+                let last_non_whitespace = self.0.char_indices().find(|pair| pair.0 > idx);
```

Comment:
This unnecessarily goes though the whole string from the start to `idx`. I think something like this would also work and be more efficient. The variable names are also slightly wrong (but already were before this PR).

<s>

```rust
            Some(last_non_whitespace) => {
                let after_last_non_whitespace = last_non_whitespace +
                    // `unwrap` never panics here because `rfind` found at least one char.
                    self.0[last_non_whitespace..].chars().next().unwrap().len_utf8();
                self.0.truncate(after_last_non_whitespace)
```

</s>

Reply:
Now that I’ve typed all this, I realize this whole `match` block can be removed and better written as:

```rust
self.0.truncate(self.0.trim_end_matches(|c| c.is_ascii_whitespace()).len());

Reply:
Sorry @SimonSapin for spamming your comment, I'm trying to understand servo I have on (maybe stupid :-)) question. With `trim_end_matches` a `&str` is created and the length of it is used. Would it also be possible to use `trim_matches` and replace the `String` member of the `DOMString` with the resulting `&str` or is the member used also somewhere else?

Reply:
`self.0 = trim_matches(…).to_string()` would work, but it makes a new memory allocation and copies the preserved part of the string. This algorithm reuses the existing allocation and, if `first_non_whitespace` happens to be zero, does not need to do any copying.